model/task_lifecycle.go

```diff
@@ -1570,24 +1570,25 @@ func resetStrandedTask(t *task.Task) error {
 		// If the task has already exceeded the unschedulable threshold, we
 		// don't want to restart it, so just mark it as finished.
 		if t.DisplayOnly {
-			for _, etID := range t.ExecutionTasks {
-				var execTask *task.Task
-				execTask, err := task.FindOneId(etID)
-				if err != nil {
-					return errors.Wrap(err, "finding execution task")
-				}
-				if execTask == nil {
-					return errors.New("execution task not found")
-				}
-				if err := MarkEnd(execTask, evergreen.MonitorPackage, time.Now(), &t.Details, false); err != nil {
-					return errors.Wrap(err, "marking execution task as ended")
+			execTasks, err := task.FindAll(db.Query(task.ByIds(t.ExecutionTasks)))
+			if err != nil {
+				return errors.Wrap(err, "finding execution tasks")
+			}
+			if len(execTasks) == 0 {
```

Comment:
We basically check with the old code, since if t.ExecutionTasks is length 0 we just skip the loop, so I think we can drop this check (cc @bynn)

Reply:
ah makes sense